ReactiveUI/Cocoa/CommonReactiveSource.cs

```diff
@@ -278,30 +234,74 @@ void resetupAll(IReadOnlyList<TSectionInfo> newSectionInfo)
             //
             // TODO: Instead of listening to Changed events and then reseting,
             // we could listen to more specific events and avoid some reloads.
-            disp.Add(sectionChanging.Subscribe(_ => {
+            disp.Add(sectionChanging.ObserveOn(ImmediateScheduler.Instance).Subscribe(_ => {
```

Comment:
Yeah, these are no-ops - I like the idea of being explicit for sure, but just to save the minor CPU time / stack inflation, let's document it via comments instead of via code